internal/test/e2e/uniqueip.go

```diff
@@ -0,0 +1,25 @@
+package e2e
+
+import (
+	"github.com/aws/eks-anywhere/pkg/logger"
+	"github.com/aws/eks-anywhere/pkg/networkutils"
+)
+
+var (
+	usedIPs = make(map[string]bool)
```

Comment:
This is not thread safe :(
Is there a way we can do this before fanning out into multiple go routines? And just pass the ip to each "worker"

If not, we can get around it using a rw mutex
However, I would suggest not using a global variable. Just encapsulate it in a type and pass the same struct to the workers

Reply:
I have moved the logic for generating IPs before threads come into play